modules/install/src/main/scala/coursier/install/InstallDir.scala

```diff
@@ -370,7 +370,7 @@ import scala.util.control.NonFatal
               else
                 baseNativePreamble
                   .withKind(Preamble.Kind.Sh)
-                  .withCommand(""""$(cd "$(dirname "$0")"; pwd)/""" + auxName(dest0.getFileName.toString, "") + "\"") // FIXME needs directory
+                  .withCommand(""""$(dirname "$(readlink -f "$0" || realpath "$0" || stat -f%Y "$0" )")/""" + auxName(dest0.getFileName.toString, "") + "\"") // FIXME needs directory
```

Comment:
On macOS, it seems the failing `readlink` prints some junk:
```text
$ sbt cli/pack
$ modules/cli/target/pack/bin/coursier install --install-dir test-install cs
$ ./test-install/cs --help
readlink: illegal option -- f
usage: readlink [-n] [file ...]
Coursier 2.0.9
Usage: cs [options] [command] [command-options]
…
```
Some `2>/dev/null` should be needed, I guess.